src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs

```diff
@@ -84,6 +91,100 @@ public HttpClient CreateClient()
                 handler.ServerCertificateCustomValidationCallback = (req, cert, chain, errors) => true;
 #endif
             }
+            // If schannel is the TLS backend, custom certificate usage must be explicitly enabled
+            else if (!string.IsNullOrWhiteSpace(_settings.CustomCertificateBundlePath) &&
+                ((_settings.TlsBackend != TlsBackend.Schannel) || _settings.UseCustomCertificateBundleWithSchannel))
+            {
+                string certBundlePath = _settings.CustomCertificateBundlePath;
+                _trace.WriteLine($"Custom certificate verification has been enabled with certificate bundle at {certBundlePath}");
+
+                // Throw exception if cert bundle file not found
+                if (!_fileSystem.FileExists(certBundlePath))
+                {
+                    throw new FileNotFoundException($"Custom certificate bundle not found at path: {certBundlePath}", certBundlePath);
+                }
+
+                Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> validationCallback = (cert, chain, errors) =>
+                {
+                    // Fail immediately if there are non-chain issues with the remote cert
+                    if ((errors & ~SslPolicyErrors.RemoteCertificateChainErrors) != 0)
+                    {
+                        return false;
+                    }
+
+                    // Import the custom certs
+                    X509Certificate2Collection certBundle = new X509Certificate2Collection();
+                    certBundle.Import(certBundlePath);
+
+                    try
+                    {
+                        // Add the certs to the chain
+                        chain.ChainPolicy.ExtraStore.AddRange(certBundle);
+
+                        // Rebuild the chain
+                        if (chain.Build(cert))
+                        {
+                            return true;
+                        }
+                        else if (chain.ChainStatus.Any(status => status.Status != X509ChainStatusFlags.UntrustedRoot))
```

Comment:
nit: don't need the `else` here since we'd have exited the block if this wasn't true.

```suggestion

                       if (chain.ChainStatus.Any(status => status.Status != X509ChainStatusFlags.UntrustedRoot))
```

Reply:
Updated - was able to simplify this by rearranging some of the conditions as well!